lib/veri_helpers.ml

```diff
@@ -0,0 +1,47 @@
+open Core_kernel.Std
+open Bap.Std
+open Bap_traces.Std
+
+module Dis = Disasm_expert.Basic
+
+(** [name_of_insns insns] - returns a name of last instruction in [insns]  *)
+let name_of_insns insns = 
+  match List.hd (List.rev insns) with
+  | None -> Or_error.error_string "instruction hasn't name"
+  | Some (_, insn) -> Ok (Insn.(name (of_basic insn)))
+
+let insns_of_mem dis mem = 
```

Comment:
Well, code-exec is by definition and execution of one instruction. Why do you introduce a list of instructions here? If a real CPU from which we got the trace, executed this memory chunk as one instruction, and we manage to disassemble it as a sequence of instructions, then it is definitely an error in the dissassembler, and there is no point to try to do with this anything more. We should bail out asap, i.e., skip to the next instruction. Here we failed drastically.